docs/specs/openapi.yaml

```
@@ -70,15 +154,120 @@ paths:
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/PostServices"
-              examples:
-                accepted:
-                  value:
-                    type: async
-                    status-code: 202
-                    status: Accepted
-                    change: "1"
-                    result: null
+                $ref: "#/components/schemas/PostServicesResponse"
+              example:
+                {
+                  "type": "async",
+                  "status-code": 202,
+                  "status": "Accepted",
+                  "change": "29",
+                  "result": null
+                }
+  /v1/logs:
+    get:
+      summary: Get service logs
+      tags:
+        - services
+      description: |
+        Fetch previously-written logs from the given services,
+        or request logs from the given services and follows them, in JSON lines format.
+      parameters:
+        - name: services
+          in: query
+          description: |
+            Service name to filter logs by.
+
+            To get logs from multiple services, specify this parameter multiple times.
+
+            If not set, returns logs for all services.
+          schema:
+            type: string
+        - name: follow
+          in: query
+          description: |
+            Whether to follow the logs (keep streaming new logs).
+
+            Must be "true" or "false"; defaults to false.
+          schema:
+            type: string
+            enum: ["true", "false"]
+            default: "false"
+        - name: n
+          in: query
+          description: |
+            Number of log entries to retrieve.
+
+            If `follow` is true, this parameter is ignored and defaults to zero.
+
+            In non-follow mode, the default value is server-defined, currently 30.
+
+            If `n` is -1, all available logs are returned (up to a server-defined limit).
+
+            If `n` is 0 and `follow` is false, a server-defined default number of logs is returned.
+
+            If `n` is a positive integer, up to that many logs are returned.
```

I found this a little bit confusing. How about using this (if it's correct):

```suggestion
            If `follow` is true, `n` is ignored.

            If `follow` is false:
            - If `n` is -1, all available logs are returned (up to a server-defined limit).
            - If `n` is 0 or not specified, a server-defined default number of logs is returned. The default is currently 30.
            - If `n` is a positive integer, up to that many logs are returned.
```